src/Microsoft.TestPlatform.CrossPlatEngine/TestSession/TestSessionPool.cs

```diff
@@ -198,7 +198,7 @@ public virtual bool ReturnProxy(TestSessionInfo testSessionInfo, int proxyId)
             // Try re-enqueueing the specified proxy.
             return sessionManager.EnqueueProxy(proxyId);
         }
-        catch (InvalidOperationException ex)
+        catch (Exception ex)
```

Comment:
We do indeed need to catch generic exceptions here. I'll add a comment in the code with the same reasoning I'm sharing right now:

Returning a proxy is more like a fire-and-forget kind of operation, it's no biggie if we aren't able to return it for whatever reason. In fact, this happens on a regular basis since there's 2 distinct calls to `ProxyDiscoveryManager/ProxyExecutionManager.Close()`, one when handling the raw message, and one when processing an actual termination message. Only one of those operations will succeed, the other one will fail, and there's no way around it without an architectural change. That's why we used to catch only `InvalidOperationException`, to account for the edge case outlined above. However, things changed when we added the fallback capability for test sessions (i.e. if we cannot match the discovery/run criteria with the criteria from one active test session that already has pre-started testhosts, then we'll just spawn testhosts on demand). The on-demand testhosts have the same regime as older testhosts, i.e. they just die after the operation they were spawned for has completed. The difference between those testhosts and the ones that are managed by a test session is that they have "-1" for the Id, and if by mistake we attempt to return one of those "non-managed" testhosts to a test session, we throw an exception that will not be caught by this method if we only handle `InvalidOperationException`. As such, rather then rely on the caller to perform sanity checks, we make sure such an edge case will not throw instead.

Reply:
Fixed, added the comment.